orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/ContextParameterProcessor.groovy

```diff
@@ -232,10 +275,43 @@ class ContextParameterProcessor {
 
       def m = new ArrayList<Method>(Arrays.asList(methods))
       m.removeAll(rejectedMethods)
+      m = m.findAll { TypeWhitelist.isAllowedForReturn(it.returnType) }
 
       return m.toArray(new Method[m.size()])
     }
   }
+
+  @CompileStatic
+  @Slf4j
+  static class FilteredPropertyAccessor extends ReflectivePropertyAccessor {
+    @Override
+    protected Method findGetterForProperty(String propertyName, Class<?> clazz, boolean mustBeStatic) {
+      Method getter = super.findGetterForProperty(propertyName, clazz, mustBeStatic)
+      if (getter && TypeWhitelist.isAllowedForReturn(getter.returnType)) {
+        return getter
+      }
+
+      if (getter) {
+        log.info("found getter for requested $propertyName but rejected due to return type $getter.returnType")
```

Comment:
if this turns out to be too verbose it could get lowered to debug but may be useful in helping people troubleshoot if they have crafted invalid expressions that are failing in a non obvious way